db/database.go

```diff
@@ -408,6 +412,9 @@ func NewDatabaseContext(dbName string, bucket base.Bucket, autoImport bool, opti
 		}
 		base.Infof(base.KeyAll, "Using metadata purge interval of %.2f days for tombstone compaction.", float64(context.PurgeInterval)/24)
 
+		if context.Options.AutoCompact {
+			context.backgroundTask("Compact database", func() { context.Compact() }, time.Duration(context.PurgeInterval)*time.Hour)
```

Comment:
I'm not sure if this wrapping of `func() {}` is necessary - I could have missed something, or just had a bit of an IDE flap...

Reply:
as long as the `Compact()` function signature matches (i.e. no return value, and no params) you should be able to specify `context.Compact` as a first-class function (without the `()`)

Reply:
Ahhh! That makes sense.